### PR TITLE
[Good First Issue][TFL FE]: Support GELU operation for TensorFlow Lite

### PR DESCRIPTION
**Pull request by Shubham-Sahoo**
### Details:
 - Supported the ```GELU``` operation for Tensorflow Lite

### Tickets:
 - https://github.com/openvinotoolkit/openvino/issues/22964

### CODE REVIEW

File: src/frontends/tensorflow_lite/src/op/gelu.cpp [hunk 1]
```diff
@@ -0,0 +1,27 @@
+// Copyright (C) 2024 Intel Corporation
```
Comment (rkazants) [inline src/frontends/tensorflow_lite/src/op/gelu.cpp:1]: please remove this file. It looks not needed. That translator was added to `tensorflow_common`

File: src/frontends/tensorflow_common/src/op/gelu.cpp [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+namespace frontend {
+namespace tensorflow {
+namespace op {
+
+OutputVector translate_gelu_op(const NodeContext& node) {
+    default_op_checks(node, 1, {"GELU"});
+    auto x = node.get_input(0);
+
+    // update these lines for best translation
+    auto approximate = node.get_attribute<GeluApproximationMode>("approximate");
```
Comment (rkazants) [inline src/frontends/tensorflow_common/src/op/gelu.cpp:22]:
```suggestion
    auto approximate = node.get_attribute<bool>("approximate", false);
```

File: src/frontends/tensorflow_common/src/op/gelu.cpp [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    auto res = make_shared<v7::Gelu>(x, approximate);
```
Comment (rkazants) [inline src/frontends/tensorflow_common/src/op/gelu.cpp:23]: based on approximate value, you should create `approximation_mode` for `v7::Gelu`

File: src/frontends/tensorflow_common/src/op/gelu.cpp [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    //
```
Comment (rkazants) [inline src/frontends/tensorflow_common/src/op/gelu.cpp:24]:
```suggestion
```

File: src/frontends/tensorflow_lite/src/op/gelu.cpp (hunk 1 above)
Comment (jane-intel) [reply src/frontends/tensorflow_lite/src/op/gelu.cpp:1]: This file is needed to take care of the TFL attribute. It should stay

File: src/frontends/tensorflow_common/include/common_op_table.hpp [hunk 5]
```diff
@@ -177,6 +177,7 @@ OP_CONVERTER(translate_x_div_y_op);
 OP_CONVERTER(translate_xlog1py_op);
 OP_CONVERTER(translate_xlogy_op);
 OP_CONVERTER(translate_zeros_like_op);
+OP_CONVERTER(translate_gelu_op);
```
Comment (keyonjie) [inline src/frontends/tensorflow_common/include/common_op_table.hpp:180]: keep it in alphabetical order.

File: src/frontends/tensorflow_common/src/op/gelu.cpp [hunk 6]
```diff
@@ -0,0 +1,32 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "common_op_table.hpp"
+#include "openvino/op/constant.hpp"
```
Comment (keyonjie) [inline src/frontends/tensorflow_common/src/op/gelu.cpp:6]: the constant.hpp is not needed?

File: src/frontends/tensorflow_lite/src/op_table.cpp [hunk 7]
```diff
@@ -195,6 +195,7 @@ std::map<std::string, CreatorFunction> get_supported_ops() {
         {"WHERE", OP_CONVERT_TYPE_RENAME(translate_where_op, "Where")},
         {"WHILE", while_op},
         {"ZEROS_LIKE", DEQUANTIZE_INPUTS(translate_zeros_like_op)},
+        {"GELU", DEQUANTIZE_INPUTS(gelu)},
```
Comment (keyonjie) [inline src/frontends/tensorflow_lite/src/op_table.cpp:198]: move it to line 89.

File: src/frontends/tensorflow_lite/src/op_table.hpp [hunk 8]
```diff
@@ -62,6 +62,7 @@ TFL_OP_CONVERTER(transpose_conv);
 TFL_OP_CONVERTER(unique);
 TFL_OP_CONVERTER(unpack);
 TFL_OP_CONVERTER(while_op);
+TFL_OP_CONVERTER(gelu);
```
Comment (keyonjie) [inline src/frontends/tensorflow_lite/src/op_table.hpp:65]: alphabet sequence.

**Review by keyonjie — COMMENTED** [review]
Thank you for addressing this, I have similar changes locally which verified work, but feel free to go ahead with it.

File: src/frontends/tensorflow_lite/src/op/gelu.cpp [hunk 9]
```diff
@@ -1,4 +1,4 @@
-// Copyright (C) 2018-2023 Intel Corporation
+// Copyright (C) 2024 Intel Corporation
```
Comment (keyonjie) [inline src/frontends/tensorflow_lite/src/op/gelu.cpp:1]: We should squash this 2nd commit to the 1st one, maybe just split the layer test if you want to.

File: src/frontends/tensorflow_common/include/common_op_table.hpp (hunk 5 above)
Comment (rkazants) [reply src/frontends/tensorflow_common/include/common_op_table.hpp:180]: this is minor, not blocker

File: src/frontends/tensorflow_lite/src/op_table.cpp (hunk 7 above)
Comment (rkazants) [reply src/frontends/tensorflow_lite/src/op_table.cpp:198]: this is minor, not blocker

File: src/frontends/tensorflow_lite/src/op_table.hpp (hunk 8 above)
Comment (rkazants) [reply src/frontends/tensorflow_lite/src/op_table.hpp:65]: this is minor, not blocker

File: tests/layer_tests/tensorflow_lite_tests/test_tfl_Unary.py [hunk 10]
```diff
@@ -34,6 +34,7 @@
     {'op_name': 'SQRT', 'op_func': tf.math.sqrt, 'kwargs_to_prepare_input': 'positive'},
     {'op_name': 'SQUARE', 'op_func': tf.math.square},
     {'op_name': 'TANH', 'op_func': tf.math.tanh},
+    {'op_name': 'GELU', 'op_func': partial(tf.nn.gelu, approximate=True)},
```
Comment (rkazants) [inline tests/layer_tests/tensorflow_lite_tests/test_tfl_Unary.py:37]: please add test case with `approximate=False`

File: src/frontends/tensorflow_lite/src/op/gelu.cpp (hunk 1 above)
Comment (rkazants) [reply src/frontends/tensorflow_lite/src/op/gelu.cpp:1]: @jane-intel, you are correct. Attribution helper is indeed needed for TF Lite, I forgot. Thanks for catch-up:)

**Comment by keyonjie** [comment]
@rkazants @jane-intel I create a new PR https://github.com/openvinotoolkit/openvino/pull/23669 which addressed comments and the layer test issue with preserving credits from @Shubham-Sahoo .

**Comment by Shubham-Sahoo** [comment]
> @Shubham-Sahoo, please apply comments. You did good job.

Done @rkazants

File: tests/layer_tests/tensorflow_lite_tests/test_tfl_Unary.py (hunk 10 above)
Comment (Shubham-Sahoo) [reply tests/layer_tests/tensorflow_lite_tests/test_tfl_Unary.py]: Added

File: src/frontends/tensorflow_lite/src/op_table.hpp (hunk 8 above)
Comment (Shubham-Sahoo) [reply src/frontends/tensorflow_lite/src/op_table.hpp]: Updated

File: src/frontends/tensorflow_lite/src/op_table.cpp (hunk 7 above)
Comment (Shubham-Sahoo) [reply src/frontends/tensorflow_lite/src/op_table.cpp]: Updated

File: src/frontends/tensorflow_common/include/common_op_table.hpp (hunk 5 above)
Comment (Shubham-Sahoo) [reply src/frontends/tensorflow_common/include/common_op_table.hpp]: Updated

File: src/frontends/tensorflow_common/src/op/gelu.cpp (hunk 3 above)
Comment (Shubham-Sahoo) [reply src/frontends/tensorflow_common/src/op/gelu.cpp]: Done

File: src/plugins/intel_cpu/thirdparty/ComputeLibrary
Comment (rkazants) [inline src/plugins/intel_cpu/thirdparty/ComputeLibrary:1]: please revert unneeded change

File: src/plugins/intel_cpu/thirdparty/onednn
Comment (rkazants) [inline src/plugins/intel_cpu/thirdparty/onednn:1]: please revert unneeded change

File: src/plugins/intel_gpu/thirdparty/onednn_gpu
Comment (rkazants) [inline src/plugins/intel_gpu/thirdparty/onednn_gpu:1]: please revert unneeded change

File: src/plugins/intel_cpu/thirdparty/ComputeLibrary
Comment (Shubham-Sahoo) [reply src/plugins/intel_cpu/thirdparty/ComputeLibrary:1]: Reverted

File: src/plugins/intel_cpu/thirdparty/onednn
Comment (Shubham-Sahoo) [reply src/plugins/intel_cpu/thirdparty/onednn:1]: Reverted

File: src/plugins/intel_gpu/thirdparty/onednn_gpu
Comment (Shubham-Sahoo) [reply src/plugins/intel_gpu/thirdparty/onednn_gpu:1]: Reverted

**Pull request closed without merging**